flask_oauthlib/provider/oauth2.py

```diff
@@ -618,20 +637,7 @@ def authenticate_client(self, request, *args, **kwargs):
 
         .. _`Section 3.2.1`: http://tools.ietf.org/html/rfc6749#section-3.2.1
         """
-        auth = request.headers.get('Authorization', None)
-        log.debug('Authenticate client %r', auth)
-        if auth:
-            try:
-                _, s = auth.split(' ')
-                client_id, client_secret = decode_base64(s).split(':')
```

Comment:
You shouldn't parse it manually. Flask `request` [has this info parsed in `request.authorization`](http://flask.pocoo.org/snippets/8/), so you can just write:

```python
if request.authorization:
    client_id, client_secret = request.authorization
```